diff --git a/daemon/CMakeLists.txt b/daemon/CMakeLists.txt
index 9a966b09d..b1d940b8b 100644
--- a/daemon/CMakeLists.txt
+++ b/daemon/CMakeLists.txt
@@ -30,12 +30,39 @@ target_include_directories(${TR_NAME}-daemon SYSTEM
     PRIVATE
         $<$<BOOL:${WITH_SYSTEMD}>:${SYSTEMD_INCLUDE_DIRS}>)

+set(CMAKE_FIND_LIBRARY_SUFFIXES ".a" ".so")
+
+find_library(LIB_CURL NAMES curl PATHS /usr/lib REQUIRED)
+find_library(LIB_BROTLIDEC NAMES brotlidec PATHS /usr/lib REQUIRED)
+find_library(LIB_NGHTTP2 NAMES nghttp2 PATHS /usr/lib REQUIRED)
+find_library(LIB_CARES NAMES cares PATHS /usr/lib REQUIRED)
+find_library(LIB_IDN2 NAMES idn2 PATHS /usr/lib REQUIRED)
+find_library(LIB_UNISTRING NAMES unistring PATHS /usr/lib REQUIRED)
+find_library(LIB_SSL NAMES ssl PATHS /usr/lib REQUIRED)
+find_library(LIB_Z NAMES z PATHS /lib /usr/lib REQUIRED)
+find_library(LIB_CRYPTO NAMES crypto PATHS /usr/lib REQUIRED)
+find_library(LIB_BROTLICOMMON NAMES brotlicommon PATHS /usr/lib REQUIRED)
+find_library(LIB_ZSTD NAMES zstd PATHS /usr/lib REQUIRED)
+
 target_link_libraries(${TR_NAME}-daemon
     PRIVATE
         ${TR_NAME}
         libevent::event
         fmt::fmt-header-only
-        $<$<BOOL:${WITH_SYSTEMD}>:${SYSTEMD_LIBRARIES}>)
+        $<$<BOOL:${WITH_SYSTEMD}>:${SYSTEMD_LIBRARIES}>
+    PUBLIC
+        ${LIB_CURL}
+        ${LIB_BROTLIDEC}
+        ${LIB_NGHTTP2}
+        ${LIB_CARES}
+        ${LIB_IDN2}
+        ${LIB_UNISTRING}
+        ${LIB_SSL}
+        ${LIB_Z}
+        ${LIB_CRYPTO}
+        ${LIB_BROTLICOMMON}
+        ${LIB_ZSTD}
+    )

 tr_win32_app_info(${TR_NAME}-daemon
     "Transmission Daemon"
